husky_components/url-input/main.js

```diff
@@ -0,0 +1,232 @@
+                items: this.items
+            }));
+        },
+
+        /**
+         * Bind necessary dom events.
+         */
+        bindDomEvents: function() {
+            this.sandbox.dom.on(
+                this.$find('.' + constants.linkClass),
+                'click',
+                this.linkHandler.bind(this)
+            );
+
+            this.sandbox.dom.on(
+                this.$find('.' + constants.specificPartClass),
+                'focusout',
+                this.specificPartChangedHandler.bind(this)
+            );
+
+            this.sandbox.dom.on(
+                this.$el,
+                constants.dataChangedEvent,
+                this.dataChangedHandler.bind(this)
+            );
+        },
+
+        /**
+         * Handles click on scheme.
+         */
+        linkHandler: function() {
+            var data = this.getData();
+            if (!data.url) {
+                return false;
+            }
+        },
+
+        /**
+         * Handles input changed.
+         */
+        specificPartChangedHandler: function(e) {
+            this.sandbox.dom.stopPropagation(e);
+
+            var $element = this.$find('.' + constants.specificPartClass),
+                specificPart = $element.val(),
+                match = urlValidator.match(specificPart, this.options.schemes),
+                data = {specificPart: specificPart};
+
+            if (!!match) {
+                data.scheme = match.scheme;
+                data.specificPart = match.specificPart;
+            }
+
+            this.setData(data);
+            this.dataChangedHandler();
+        },
+
+        /**
+         * Handles data changed.
+         */
+        dataChangedHandler: function() {
+            var data = this.getData();
+            this.$find('.' + constants.specificPartClass).val(data.specificPart);
+            this.$find('.' + constants.schemeClass).html(data.scheme);
+            this.$find('.' + constants.schemeClass).attr('href', data.url);
+        },
+
+        /**
+         * Bind necessary aura events.
+         */
+        bindCustomEvents: function() {
+            this.sandbox.on(
+                'husky.dropdown.' + this.options.instanceName + '.item.click',
+                this.selectSchemeHandler.bind(this)
+            );
+        },
+
+        /**
+         * Handles scheme dropdown value click.
+         *
+         * @param {{id, name}} item
+         */
+        selectSchemeHandler: function(item) {
+            this.setData({scheme: item.id});
+
+            this.dataChangedHandler();
+        },
+
+        /**
+         * Set data to dom-data and generate URL.
+         *
+         * @param {{scheme, specificPart}} data
+         */
+        setData: function(data) {
+            var newData = this.sandbox.util.extend(true, {}, this.getData(), data);
+
+            newData.url = null;
+            if (!!newData.scheme && !!newData.specificPart) {
+                newData.url = this.getUrl(newData);
```

Comment:
I would not write this information on this object, since at can be recreated out of the other two properties at any time. Duplicating this information just raises the probability that an error occurs.